app/Models/Beatmapset.php

```diff
@@ -545,11 +545,19 @@ public function setApproved($state, $user)
             $this->approvedby_id = null;
         }
 
+        $beatmaps = $this->beatmaps();
```

Comment:
I did only say "before saving" but there's no point setting all the attributes just to throw them away in a semi-unrelated parameter validation :thinking:

Reply:
Yeah, I'm not sure what was I thinking when code this in the night :thinking:

Reply:
Should `$beatmaps->update(...)` before or after `$this->save()`?

Reply:
either should be fine